src/Microsoft.AspNetCore.Routing/Matchers/MatcherBase.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    internal abstract class MatcherBase : Matcher
+    {
+        protected internal abstract void SelectCandidates(HttpContext httpContext, ref CandidateSet candidates);
+
+        public sealed unsafe override Task MatchAsync(HttpContext httpContext, IEndpointFeature feature)
+        {
+            if (httpContext == null)
+            {
+                throw new ArgumentNullException(nameof(httpContext));
+            }
+
+            if (feature == null)
+            {
+                throw new ArgumentNullException(nameof(feature));
+            }
+
+            var path = httpContext.Request.Path.Value;
+            var buffer = stackalloc PathSegment[32];
```

Comment:
Move array size to const. Add comment explaining why 32

Reply:
Fixed